docs/user-manual/en/upgrading.md

```diff
@@ -48,3 +48,23 @@ most cases_ the instance can be upgraded to a newer version simply by changing
 the value of this property to the location of the new broker home. Please refer
 to the aforementioned [versions](versions.md) document for additional upgrade
 steps (if required).
+
+It is also possible to do these steps automatically as it can be seen in the next section.
+
+## Upgrading tool
+
+An automatic approach can be used to upgrade the instance. You may simply call `./artemis upgrade <old-instance>`.
+
+```shell
+cd $NEW_ARTEMIS_DOWNLOAD/bin/
+./artemis upgrade PATH_TO_UPGRADING_INSTANCE
+```
+
+artemis, artemis.profile will be updated to the new version.
+
+The tool will also update log4j2.properties (if you are migrating from a version previous to 2.27.0).
+
+> **Note:**
+> 
+>to avoid removing eventual user's customizations, the upgrade process won't automatically bring new additions to the scripts itself. Please compare your artemis and artemis.profile to newer versions after the upgrade.
```

Comment:
I think this is over-selling the 'automatation', given the note at the end. The description here isnt clear as to what it will do. I think a user will struggle to know what this means until trying it and finding it doenst necessarily do what they expected.

To have a new version file to compare with, a user would still also need to 'artemis create' a new separate broker instance to see its config, but now thats in addition to running the 'artemis upgrade' command on their existing one to take care of some of the updates, but not others. This isnt all that clear from the 'upgrade' name for the command, or this documentation.

That was one good thing about the previous 'create new files' approach. Theyd already have the new file, and theyd have their backed up old file to compare it with and find/port port their own modifications across if needed.

Reply:
I could just always replace the ./artemis script though. it's one thing I was avoiding to do.

Reply:
@gemmellr I will make one fundamental change here...


I will do the opposite.. always replace the whole script... but saving the locations... and telling the user to compare the script with the former backed up for any special adjustments they have done themselves to the scripts.

Reply:
I think that ultimately works better for the user, and shoudl be simpler to implement. Means the user just has one command to run, and they can see all the differences to their earlier config and pick our any of their own customisations they still want.

Reply:
I have already changed it to keep stuff... 

it's not ready yet to merge.. but I already pushed the changes... (I will make the PR as a draft until I finish it)